### PR TITLE
Refactor subworkflows

### PR DESCRIPTION
**Pull request by ewels**
* Don't align different subworkflows under the same name
* Don't use `params` within subworkflows, instead use `take`

### CODE REVIEW

**Review by drpatelh — APPROVED** [review]
🚀 Generally nice to have a separate sub-workflow that deals with all of the genome files because it takes all of that logic out of the main script. And other than listing all outputs in the `emit` block LGTM!

**Comment by ewels** [comment]
> Generally nice to have a separate sub-workflow that deals with all of the genome files because it takes all of that logic out of the main script.

I started doing this already twice but got myself in a right tangle and ended up with code that I found much more difficult to follow 😅  In the end I gave up as it's only 3 processes and figured if I was struggling with it then others would too. Might become more useful if we start adding more aligners though, as planned.

If it's ok I'll take that as a future task.

File: subworkflows/local/prepare_genome.nf [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            versions = versions.mix(SAMTOOLS_FAIDX.out.versions)
+        }
+    }
+
+    emit:
+    fasta = fasta
+    bismark_index = bismark_index
+    bwameth_index = bwameth_index
+    fasta_index = fasta_index
+    versions = versions
```
[the remark this answers is not in the document]
Comment (ewels) [reply subworkflows/local/prepare_genome.nf:70]: I was wondering if I could do this but I got scared! They're still available as named outputs then?

[the remark this answers is not in the document]
Comment (maxulysse) [reply subworkflows/local/prepare_genome.nf]: sorry if it was a bad idea...

[the remark this answers is not in the document]
Comment (ewels) [reply subworkflows/local/prepare_genome.nf]: I didn't try it earlier, was just guessing based on what I saw in the Nextflow docs.

Tests are passing, so I think you're right 👍🏻 Nice!

**Pull request merged**